### ISSUE TITLE
[Proposal] Surface "critical" changes that happen in `dapr` weekly to the community

### ISSUE THREAD

**Issue by mukundansundar** [dapr/community#234]
With the growth of `dapr` project and the community, there are multiple changes that happen in parallel across multiple repos in the `dapr` project. 
Not all changes can be followed by anyone. This proposal is to see if we can provide a concise summary of changes say `new proposal created`, `proposal accepted`, critical changes made to  runtime/contrib or SDKs e.g.: `addition of a new feature`, `changes in preexisting criteria(say new requirements needed for a component to be certified stable)`.

 

As it exists right now most of the conversations happen across multiple repos and in either different issues or PRs. Also there are certain conversations that happen in discord that are then reflected as PR contributions in `dapr`.

 

Keeping track of all these changes/ going back and being able to search for some important issue that has been agreed upon and closed etc. is hard.

 

This proposal is to identify categories of the information that should be surfaced to the community on a weekly basis that provides visibility into changes happening in the project.

 

Some of the categories can be tracked and surfaced on a weekly basis :
1. `New API` code changes 
2. `New criteria/process definition`
3. `Proposals in review`, `Accepted. proposals`
4. "Large", "impactful" code change in runtime repo -- can be a large refactor, internal behavior change etc. based on pre-identified critical paths
  - `pkg/http`, `pkg/diagnostics` change in runtime
  - `tests/conformance` change in components-contrib repo
5. `Breaking changes` that will be part of the next milestone

 

These are just some broad categories by which we can identify changes in dapr project. We can add action commands that maintainers and approvers can use to tag certain PRs that should be surfaced. We can also automate the `daprbot` to identify proposals, acceptance of the same, changes of files in certain paths of the repo which have been pre-identified as critical code paths etc.

 

`daprbot` can potentially be augmented to post a weekly summary defined under these categories to a channel in Discord.

 

Benefits : 
1. Provide a broader visibility into changes happening across the project
2. Provide visibility into tracking progress of milestone issues 
3. Provide a single place to track broad impactful changes to the project
4. Provide visibility to the community
5. Provide visibility of repo maintainers of changes that happen in other repos that might impact the repo that they are maintainers of
 

This is no means a substitute to general everyday triaging, pr reviews and issues reviews that is done. But it shall complement those efforts also.

 

cc @dapr/maintainers-community @dapr/maintainers-dapr Thoughts?

**Comment by ItalyPaleAle** [comment]
I think this is a good proposal that addresses a real need of the community.

As for its execution, I am not a fan of a bot just sending links on Discord. The reason is that those would lack context and could be perceived as "spammy". Just posting links without context usually leads to a very low CTR.

I would recommend that there be some sort of curation here, in which someone (maybe a maintainer) posts a message with links to the issues/PRs AND a 1-sentence explanation of why that is relevant and we need from the community. For example, this person could write that we are sharing issue X because we are seeking feedback on a proposal, or we are highlighting possible breaking changes, or... This will also highlight why it's important for the community to chime in.

This way, it feels more personal, more actionable, and will see higher CTR and better engagement.

**Comment by nyemade-uversky** [comment]
> I would recommend that there be some sort of curation here, in which someone (maybe a maintainer) posts a message with links to the issues/PRs AND a 1-sentence explanation of why that is relevant and we need from the community. For example, this person could write that we are sharing issue X because we are seeking feedback on a proposal, or we are highlighting possible breaking changes, or... This will also highlight why it's important for the community to chime in.


I like this proposal and I agree with this more curated way of posting the updates. Without context this information could be confusing to other community members. I'd be happy to work out a place to post this info and take ownership for getting it shared w/ the community. 

@msfussell, @dapr/maintainers-community  Any thoughts here?

**Comment by mukundansundar** [comment]
@ItalyPaleAle I can understand that providing more context for some the changes would be great. And @nyemade-uversky Thanks for volunteering for this. 
If we can build in more automation for the context also, that would be great. But we can start with manually curating it for now. 
@dapr/maintainers-community @dapr/maintainers-dapr Any thoughts on proceeding forward with this?

**Comment by akhilac1** [comment]
@mukundansundar  @johnewart - where do preview features come in. Can the criteria be included in the proposal?

**Comment by mukundansundar** [comment]
> @mukundansundar @johnewart - where do preview features come in. Can the criteria be included in the proposal?

@akhilac1 Please add comment in dapr/proposals#1 ...

**Comment by msfussell** [comment]
@mukundansundar - What are you thoughts on keeping this issue open? Can this be resolved now?

**Comment by msfussell** [comment]
No longer relevant and addressed in new processes in weekly ship rooms
